src/components/MainNavbar.js

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { Container, Navbar, Spinner } from "react-bootstrap";
+import OneInputModal from "./OneInputModal";
+import TextLogo from "./TextLogo";
+import { useNavigate } from "react-router-dom";
+import { v4 as uuidv4 } from "uuid";
+import Peer from "peerjs";
+
+import Profile from "./Profile";
+
+// hooks
+import useUpdateinput from "../utility/useUpdateinput";
+import useGetCamera from "../utility/useGetCamera";
+
+
+export default function MainNavbar(props) {
+    const navigate = useNavigate();
+    const peer = new Peer();
+    const [target, changeInput] = useUpdateinput({ oneinput: uuidv4() });
+    const [iWantToCall, setIWantToCall] = React.useState("");
+
+    function createRoom() {
+        // room id gets set once you are on the rooms page 
+        // so others can call you 
+        navigate(`/room/${target.oneinput}`)
+    }
+
+    function targetFunc(e) {
+        setIWantToCall(e.target.value)
+    }
+
+    async function startCall() {
+        const getMedia = useGetCamera;
+
+        await navigate(`/room/${iWantToCall}`)
+        const userMedia = await getMedia()
+        const call = peer.call(iWantToCall, userMedia)
+
+        call.on("stream", async (stream) => {
+            const participant = await document.getElementById("participant")
+            participant.srcObject = stream
+        })
+    }
```

Comment:
Logic like this is used in multiple places. It seems to me that there might be a way to extract that logic into a custom hook that has `startCall`, `endCall`, ...